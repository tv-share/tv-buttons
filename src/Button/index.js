import "./button.*";
import Button from "./Button";

export default Button;