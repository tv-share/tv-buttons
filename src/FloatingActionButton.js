import React from "react";
import PropTypes from "prop-types";

import "./style/floatingActionButton";
/**
* Renders the SelectInput Component
* @param  {String}    	className is a string that will be attached to the select input's className (optional)
* @param  {String}      icon defines the path of the icon to be displayed inside the button
* @param  {Function}    onClick defines a function to be called whenever the button is pressed
* @param  {String}      fixed defines if button should be fixed in the bottom right of the screen
* @param  {Object}      events is an object with one or more React Synthetic Events
* @return {JSX}         SelectInput Component
*/
const FloatingActionButton = ({ className, icon, onClick, fixed = false, events = {} }) => {
	return (
		<div className={`tv-floating-action-button ${fixed ? "-fixed" : ""} ${className ? className : ""}`} icon={icon} onClick={onClick}>
			<img src={icon}/>
		</div>
	);
};

FloatingActionButton.propTypes = {
	icon: PropTypes.string.isRequired,
	onClick: PropTypes.func.isRequired,
	className: PropTypes.string,
	fixed: PropTypes.bool,
	events: PropTypes.object
};

export default FloatingActionButton;