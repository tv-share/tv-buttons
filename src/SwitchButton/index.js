import "./switchButton.*";
import SwitchButton from "./SwitchButton";

export default SwitchButton;