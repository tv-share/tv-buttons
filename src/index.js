import ToggleButton from "./lib/ToggleButton";

export { ToggleButton };
