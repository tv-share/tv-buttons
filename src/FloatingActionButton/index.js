import "./floatingActionButton.*";
import FloatingActionButton from "./FloatingActionButton";

export default FloatingActionButton;