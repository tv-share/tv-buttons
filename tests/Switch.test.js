import React from 'react';
import renderer from 'react-test-renderer';

import { Switch } from "../src/index";

const fn = () => { };

it('renders correctely on ON position', () => {
	const component = renderer.create(
		<Switch isEnabled={true} onClick={() => console.log("ON")}/>
	);

	expect(component.toJSON()).toMatchSnapshot();
});

it('renders correctely on OFF position', () => {
	const component = renderer.create(
		<Switch isEnabled={false} onClick={() => console.log("OFF")}/>
	);

	expect(component.toJSON()).toMatchSnapshot();
});