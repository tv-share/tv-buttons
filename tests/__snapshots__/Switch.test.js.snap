// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctely on OFF position 1`] = `
<div
  className="tv-switch "
>
  <div
    className="box -off"
    onClick={[Function]}
  >
    <div
      className="circle -off"
    />
  </div>
</div>
`;

exports[`renders correctely on ON position 1`] = `
<div
  className="tv-switch "
>
  <div
    className="box -on"
    onClick={[Function]}
  >
    <div
      className="circle -on"
    />
  </div>
</div>
`;
